src/theory/bags/bag_solver.cpp

```diff
@@ -53,11 +53,27 @@ void BagSolver::postCheck()
   }
 }
 
+set<Node> BagSolver::getElementsForBinaryOperator(const Node& n)
+{
+  set<Node> elements;
+  const set<Node>& downwards = d_state.getElements(n);
+  const set<Node>& upwards0 = d_state.getElements(n[0]);
+  const set<Node>& upwards1 = d_state.getElements(n[1]);
+
+  set_union(downwards.begin(),
+            downwards.end(),
+            upwards0.begin(),
+            upwards0.end(),
+            inserter(elements, elements.begin()));
+  elements.insert(upwards1.begin(), upwards1.end());
+  return elements;
+}
+
 void BagSolver::checkUnionDisjoint(const Node& n)
 {
   Assert(n.getKind() == UNION_DISJOINT);
-  TypeNode elementType = n.getType().getBagElementType();
-  for (const Node& e : d_state.getElements(elementType))
+  set<Node> elements = getElementsForBinaryOperator(n);
```

Comment:
```suggestion
  std::set<Node> elements = getElementsForBinaryOperator(n);
```

Reply:
Similar for other places, I prefer always having std:: for standard library structures.

Reply:
I prefer the same. I just forgot during copy/paste